src/FwdState.cc

```diff
@@ -898,7 +898,7 @@ FwdState::connectStart()
 
     assert(Config.forward_max_tries - n_tries > 0);
     HttpRequest::Pointer cause = request;
-    const auto cs = new HappyConnOpener(destinations, calls.connector, cause, start_t, al);
+    const auto cs = new HappyConnOpener(destinations, calls.connector, cause, start_t, Config.forward_max_tries - n_tries, al);
```

Comment:
I think we should pass `n_tries` here instead of effectively freezing `Config.forward_max_tries` for the entire duration of the HappyConnOpener job run. Let HappyConnOpener use the current Config.forward_max_tries value. That configuration value may change while HappyConnOpener is running (and we already support such changes in configuration AFAICT).

Properly addressing the above change request will automatically fix the n_tries description that is currently lying. Please do not forget to remove the unused maxTries member afterwords.

Reply:
resolved at d5cc891

Probably we should use two members, the n_tries and the n_forwarding_tries